python/sglang/srt/model_executor/model_runner.py

```diff
@@ -225,19 +222,19 @@ def init_torch_distributed(self):
                 backend=backend,
                 world_size=self.tp_size,
                 rank=self.tp_rank,
-                local_rank=self.gpu_id,
+                local_rank=self.tp_rank % 8,  # Assumes 8 GPUs / node.
```

Comment:
this may not be good idea for oss sglang (there are many nodes with less than 8 gpus per node in aws)

Reply:
That's right.

It turns out I misunderstood the meaning of `local_rank` here anyway, since it's used as the device to set from a local processes's perspective [here](https://github.com/vllm-project/vllm/blob/main/vllm/distributed/parallel_state.py#L198), so I think it should actually just be zero when `CUDA_VISIBLE_DEVICES` is set.